scripts/gen_graphs.py

```diff
@@ -472,35 +474,45 @@ def generate_language_summary_pie_chart(oss_entity):
 
 def generate_cost_estimates_bar_chart(oss_entity):
     """
-    This function generates a pygal bar chart for estimated costs with rounded values and a dollar sign.
+    This function generates a pygal bar chart for estimated costs 
+    with rounded values and a dollar sign.
 
     Arguments:
         oss_entity: the OSSEntity to create a graph for.
     """
 
     bar_chart = pygal.Bar(legend_at_bottom=True)
 
+    if oss_entity.metric_data is not None:
+        metric_data = oss_entity.metric_data.get('cocomo', {})
+        estimated_cost_low = metric_data.get('estimatedCost_low', 0)
+        estimated_cost_high = metric_data.get('estimatedCost_high', 0)
+    else:
+        estimated_cost_low = 0.0
+        estimated_cost_high = 0.0
 
-    metric_data = oss_entity.metric_data['cocomo']
-
-    estimatedCost_low = metric_data.get('estimatedCost_low', 0)
-    estimatedCost_high = metric_data.get('estimatedCost_high', 0)
+    formatted_estimated_cost_low = float(estimated_cost_low or 0.0)
```

Comment:
Why is this variable needed? Could you cast to a float when you are assigning `estimated_cost_low` and `estimated_cost_high`? 

It's also slightly confusing since the formatting is done later on lines 504 and 506.

I would either just reuse the same variable names or just cast to float on the same line you assign the variables.

Reply:
Refactored to add `float` directly into the first `variables estimated_cost_low` and `estimated_cost_high`. Removed `formatted_cost` variables.